pkg/client-sdk/example/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"sync"
+
+	arksdk "github.com/ark-network/ark-sdk"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	var (
+		explorerUrl = "http://localhost:3001"
+		network     = "regtest"
+		aspUrl      = "http://localhost:6000"
+		aspPubKey   = "ASP_PUBKEY"
+
+		ctx         = context.Background()
+		explorerSvc = arksdk.NewExplorer(explorerUrl)
+	)
+
+	alice, err := arksdk.NewSingleKeyWallet(
+		explorerSvc, network, "",
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	aliceArkClient, err := arksdk.New(
+		alice, explorerUrl, arksdk.Grpc, network, aspUrl, aspPubKey,
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := aliceArkClient.Connect(ctx); err != nil {
+		log.Fatal(err)
+	}
+
+	_, aliceOnchainAddr, err := aliceArkClient.Receive(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	_, err = runCommand("nigiri", "faucet", "--liquid", aliceOnchainAddr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	txID, err := aliceArkClient.Onboard(ctx, 20000)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Infof("Alice onboarded with txID: %s", txID)
+
```

Comment:
we should mint a block with `nigiri rpc --liquid generatetoaddress 1 $(nigiri rpc --liquid getnewaddress)` and check that the balance has been updated correctly

Reply:
Check [this](https://github.com/ark-network/ark/pull/193#issuecomment-2217200874) comment, i think it would be ideal if we come up with good wallet abstraction and pass it as interface than to "leave" sdk to do creation of concrete wallets

Reply:
Done